internal/controllers/builder.go

```diff
@@ -5,17 +5,17 @@ import (
 	"errors"
 	"fmt"
 
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+
 	ctrl "sigs.k8s.io/controller-runtime"
-	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 
 	"github.com/tilt-dev/tilt/internal/store"
 )
 
 type Controller interface {
 	reconcile.Reconciler
-	SetClient(client ctrlclient.Client)
-	SetupWithManager(mgr ctrl.Manager) error
+	CreateBuilder(mgr ctrl.Manager) (*builder.Builder, error)
```

Comment:
Well, specifically `FieldIndexer::IndexField()` returns an `error`: https://github.com/kubernetes-sigs/controller-runtime/blob/484f82a4a92dcb7845c331c5f7f4f86d8b772800/pkg/client/interfaces.go#L135

It's kludgy/weird here for the normal case though.

I could split this up so that the normal `Controller` interface just has `CreateBuilder(ctrl.Manager) *builder.Builder` and then an optional interface with something like `SetUpFieldIndexer(client.FieldIndexer) error` for those that need it which would be a bit more explicit

Reply:
ah ok